python/lsst/cell_coadds/_fits.py

```diff
@@ -19,11 +19,22 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+"""Module to handle FITS serialization and de-serialization.
+
+This module is under its own version control, using the module global constant
+MODULE_VERSION, and follows semantic versioning. Semantic versioning allows us
+to know which files written by an earlier version of this module is still
+readable or not. Any changes to this module, even as trivial as formatting
+changes and documentation updates should bump the version number (in this case,
+it would bump the patch). This is mandated in GitHub Actions.
+"""
```

Comment:
I don't think it's obvious how one maps semver to I/O versioning; I think we need to spell out what is and is not supported under major, minor, and patch version bumps (and we might find that those are not sufficiently granular for the guarantees we might want to make, if both forwards- and backwards-compatibility and read/write permutations are in play).

But I also think going to all that effort might be premature, and maybe a single-integer version would be better for now?  The most important thing is getting some kind of number into the files so future code doesn't have to guess about what it's dealing with.

I also have to admit that I'm a little skeptical of the value of the GitHub Action and the policy of bumping the patch version for all changes.  Is this something you've seen in other codebases?

Reply:
I'm envisioning the semantic versioning to work in exactly the same way as it does for software. I suppose one assumption here is that the version with which we are reading a file is never older than the version used to write a file, since a write operation must preceed a read operation. An example of a major bump would be if we decide to use `ImageHDU`s instead of `BinTableHDU`s and drop the support for the latter. That would be a breaking change rendering the old files unreadable. Examples of minor bump would include this PR, supporting versions with or without `VERSION` in the header and substituting a default value if not found for newer items like `day_obs` and `physical_filter` we just introduced. The changes in all of the three examples aren't fundamentally different - in one case _we decide_ to drop support, and for the others we don't.

Reply:
We still rely on the developer doing the due diligence in deciding the correct version after a change. This is also enforceable but much harder. We could also call `0.x.y` versions unstable in that we don't give any semver kind of guarantee until `1.0.0`. Bumping minor versions for every change would be equivalent to just an integer numbering that you suggested.  It seems wasteful to me to discard old data for any missing metadata. So this is definitely something we should support in production, even if not now.